site/src/xServices/users/usersXService.ts

```diff
@@ -70,393 +80,455 @@ export type UsersEvent =
-            target: "confirmUserDeletion",
-            actions: ["assignUserIdToDelete"],
-          },
-          ACTIVATE_USER: {
-            target: "confirmUserActivation",
-            actions: ["assignUserIdToActivate"],
-          },
-          RESET_USER_PASSWORD: {
-            target: "confirmUserPasswordReset",
-            actions: ["assignUserIdToResetPassword", "generateRandomPassword"],
-          },
-          UPDATE_USER_ROLES: {
-            target: "updatingUserRoles",
-            actions: ["assignUserIdToUpdateRoles"],
-          },
-        },
-      },
-      confirmUserSuspension: {
-        on: {
-          CONFIRM_USER_SUSPENSION: "suspendingUser",
-          CANCEL_USER_SUSPENSION: "idle",
         },
-      },
-      confirmUserDeletion: {
-        on: {
-          CONFIRM_USER_DELETE: "deletingUser",
-          CANCEL_USER_DELETE: "idle",
-        },
-      },
-      confirmUserActivation: {
-        on: {
-          CONFIRM_USER_ACTIVATION: "activatingUser",
-          CANCEL_USER_ACTIVATION: "idle",
-        },
-      },
-      suspendingUser: {
-        entry: "clearSuspendUserError",
-        invoke: {
-          src: "suspendUser",
-          id: "suspendUser",
-          onDone: {
-            // Update users list
-            target: "gettingUsers",
-            actions: ["displaySuspendSuccess"],
+        gettingUsers: {
+          entry: "clearGetUsersError",
+          invoke: {
+            src: "getUsers",
+            id: "getUsers",
+            onDone: [
+              {
+                target: "idle",
+                actions: "assignUsers",
+              },
+            ],
+            onError: [
+              {
+                target: "idle",
+                actions: [
+                  "clearUsers",
+                  "assignGetUsersError",
+                  "displayGetUsersErrorMessage",
+                ],
+              },
+            ],
           },
-          onError: {
-            target: "idle",
-            actions: ["assignSuspendUserError", "displaySuspendedErrorMessage"],
+          tags: "loading",
+        },
+        idle: {
+          on: {
+            SUSPEND_USER: {
+              target: "confirmUserSuspension",
+              actions: "assignUserIdToSuspend",
+            },
+            DELETE_USER: {
+              target: "confirmUserDeletion",
+              actions: "assignUserIdToDelete",
+            },
+            ACTIVATE_USER: {
+              target: "confirmUserActivation",
+              actions: "assignUserIdToActivate",
+            },
+            RESET_USER_PASSWORD: {
+              target: "confirmUserPasswordReset",
+              actions: [
+                "assignUserIdToResetPassword",
+                "generateRandomPassword",
+              ],
+            },
+            UPDATE_USER_ROLES: {
+              target: "updatingUserRoles",
+              actions: "assignUserIdToUpdateRoles",
+            },
+            UPDATE_PAGE: {
```

Comment:
UPDATE_PAGE and UPDATE_FILTER are new. UPDATE_FILTER sends RESET_PAGE to the pagination machine, which sends UPDATE_PAGE to this machine.